x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/delete_rules_route.ts

```diff
@@ -43,9 +46,17 @@ export const createDeleteRulesRoute: Hapi.ServerRoute = {
         id,
         ruleId,
       });
-
+      const ruleStatuses = await savedObjectsClient.find({
+        type: ruleStatusSavedObjectType,
+        perPage: 10,
+        search: `"${id}"`,
```

Comment:
You want `rule.id` here instead of id as it is what was retrieved by either id or rue_id.